arch/arm/core/offsets/offsets_aarch32.c

```diff
@@ -32,12 +32,15 @@
 GEN_OFFSET_SYM(_thread_arch_t, basepri);
 GEN_OFFSET_SYM(_thread_arch_t, swap_return_value);
 
-#if defined(CONFIG_USERSPACE) || defined(CONFIG_FPU_SHARING)
+#if defined(CONFIG_ARM_STORE_EXC_RETURN) || defined(CONFIG_USERSPACE)
 GEN_OFFSET_SYM(_thread_arch_t, mode);
+#endif
```

Comment:
are you using `_thread_offset_to_mode` when `CONFIG_ARM_STORE_EXC_RETURN` is defined?

Reply:
I think only when CONFIG_FPU_SHARING is defined, right @oyvindronningstad ? So, when we only care about S-NS transitions, without USERSPACE and without FPU_SHARING, we do not use the mode offset, only the mode_exc_return..

Reply:
Yep, I think that's right. Changed to 
```suggestion
#if defined(CONFIG_FPU_SHARING) || defined(CONFIG_USERSPACE)
GEN_OFFSET_SYM(_thread_arch_t, mode);
#endif
```

Reply:
thanks
@carlocaione ^^